src/Query/SqlWalker.php

```diff
@@ -1412,7 +1412,9 @@ public function walkObjectExpression(string $dqlAlias, array $partialFieldSet, s
 
             $sqlParts[] = $col . ' AS ' . $columnAlias;
 
-            $this->scalarResultAliasMap[$resultAlias][] = $columnAlias;
+            if (is_string($resultAlias)) {
```

Comment:
Let's make it explicit that this is actually a null-check.

```suggestion
            if ($resultAlias !== null) {
```